src/databricks/labs/ucx/hive_metastore/catalog_schema.py

```diff
@@ -229,18 +205,96 @@ def _validate_location(self, location: str) -> bool:
         logger.warning(f"No matching external location found for: {location}")
         return False
 
-    def _create_catalog(self, catalog: str, catalog_storage: str, *, properties: dict[str, str] | None) -> None:
-        logger.info(f"Creating UC catalog: {catalog}")
+    def _get_catalog(
+        self,
+        catalog: Catalog,
+        *,
+        timeout: dt.timedelta | None = None,
+    ) -> Catalog | None:
+        """Get a catalog.
+
+        Args:
+            catalog (Catalog) : The catalog to get.
+            timeout (dt.timedelta) : Timeout to wait before concluding the catalog does not exist. If None, no timeout
+                is applied. Defaults to `None`.
+
+        Returns:
+            Catalog : The catalog it got.
+            None : If the catalog does not exist.
+        """
+        if timeout:
+            get = retried(on=[NotFound], timeout=timeout)(self._ws.catalogs.get)
+        else:
+            get = self._ws.catalogs.get
+        try:
+            catalog_info = get(catalog.name)
+            return Catalog(catalog_info.name)
+        except (NotFound, TimeoutError):
+            return None
+        except DatabricksError as e:
+            logger.warning(f"Unexpected error when getting catalog: {catalog.name}", exc_info=e)
+            return None
+
+    def _create_catalog(
+        self,
+        catalog: Catalog,
+        catalog_storage: str,
+        *,
+        properties: dict[str, str] | None,
+    ) -> Catalog:
+        logger.info(f"Creating UC catalog: {catalog.name}")
         if catalog_storage == "metastore":
-            self._ws.catalogs.create(catalog, comment="Created by UCX", properties=properties)
+            self._ws.catalogs.create(catalog.name, comment="Created by UCX", properties=properties)
         else:
             self._ws.catalogs.create(
-                catalog,
+                catalog.name,
                 storage_root=catalog_storage,
                 comment="Created by UCX",
                 properties=properties,
             )
+        catalog_created = self._get_catalog(catalog, timeout=dt.timedelta(seconds=30))
```

Comment:
```suggestion
        catalog_created = self._get_catalog(catalog, timeout=self._timeout)
```
inject this timeout via constructor, set it to 30 seconds in the context and None in unit tests. same for other calls